src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -1562,25 +1599,60 @@ public void updateCounts( final MoleculeCounts moleculeCounts,
         readCounts.addBaseCalls(read.getLength());
 
         if ( read.isUnmapped() || read.isDuplicate() || read.failsVendorQualityCheck() ) {
-            readCounts.bumpNReadsUnmapped();
-            return ReadReport.NULL_REPORT;
+            return documentUnmappedRead(read);
         }
 
-        final Interval trim = calculateTrim(read.getBaseQualitiesNoCopy());
+        Interval trim = calculateTrim(read.getBaseQualitiesNoCopy());
         if ( trim.size() == 0 ) {
-            readCounts.bumpNLowQualityReads();
-            return ReadReport.NULL_REPORT;
+            return documentLowQualityRead(read);
+        }
+
+        // don't process past end of fragment when fragment length < read length
+        if ( read.isProperlyPaired() ) {
```

Comment:
Since `Interval trim` is being changed in this new `if`-block, I would propose absorbing this block into `calculateTrim(...)`. 
I tried to use "Refactor -> Extract -> Method", and the IDE tells me that it is impossible only because of the two statements `return documentLowQualityRead(read)`, but I think it is still do-able by returning a 0-sized interval in `calculateTrim(...)`, just like what happened in lines 1606-1608.